### PR TITLE
Adding WordPress Elastic BeanStalk Tutorial

### PR DESCRIPTION
**Pull request by oswaldderiemaecker**
Adding WordPress Elastic BeanStalk Tutorial

### CODE REVIEW

File: _posts/2017-02-18-wordpress-elastic-beanstalk.markdown [hunk 1]
```diff
@@ -0,0 +1,964 @@
...
+	"s3:PutObject",
+	"s3:PutObjectAcl",
+	"s3:PutObjectAclVersion"
+    ],
+    "Resource": "arn:aws:s3:::my-wordpress-site-backup/*",
+    "Condition": {}
+},
+{
+    "Effect": "Allow",
+    "Action": "s3:ListAllMyBuckets",
```
Comment (Pierozi) [inline _posts/2017-02-18-wordpress-elastic-beanstalk.markdown]: Did we need to list all buckets ?

File: _posts/2017-02-18-wordpress-elastic-beanstalk.markdown [hunk 2]
```diff
@@ -0,0 +1,964 @@
...
+            ]
+        },
+        {
+            "Action": [
+                "elasticbeanstalk:*"
+            ],
+            "Effect": "Allow",
+            "Resource": [
+                "arn:aws:elasticbeanstalk:*::solutionstack/*",
+                "arn:aws:elasticbeanstalk:<AWS-REGION>:<AWS-ACCOUNT-ID>:application/my-wordpress-site",
```
Comment (Pierozi) [inline _posts/2017-02-18-wordpress-elastic-beanstalk.markdown]: We need specify to user that he must replace `<AWS-REGION>:<AWS-ACCOUNT-ID>` with the good one.

File: _posts/2017-02-18-wordpress-elastic-beanstalk.markdown [hunk 3]
```diff
@@ -0,0 +1,964 @@
+---
+layout:         post
+title:          "WordPress Elastic BeanStalk Deployment"
```
Comment (ppaulis) [inline _posts/2017-02-18-wordpress-elastic-beanstalk.markdown]: It's "Beanstalk" with s in lowercase (checked on AWS Website)

File: _posts/2017-02-18-wordpress-elastic-beanstalk.markdown [hunk 4]
```diff
@@ -0,0 +1,964 @@
...
+date:           2017-02-18 15:24:11
+author:         'oswaldderiemaecker'
+image:          '/assets/2017-02-18/wordpress.png'
+hasDescription: true
+categories:     tutorial
+---
+The goal of this project is to have a WordPress environment that let you develop, test, package and deploy an immutable WordPress on different environments like staging and production.
```
Comment (ppaulis) [inline _posts/2017-02-18-wordpress-elastic-beanstalk.markdown]: that let**s** you

File: _posts/2017-02-18-wordpress-elastic-beanstalk.markdown [hunk 5]
```diff
@@ -0,0 +1,964 @@
...
+
+<!--more-->
+
+- [Introduction](#introduction)
+- [Requirements](#requirements)
+- [Set-up your Development environment](#set-up-your-development-environment)
+  - [Prepare your WordPress Repository](#prepare-your-wordpress-repository)
+  - [Dependencies management with composer](#dependencies-management-with-composer)
```
Comment (ppaulis) [inline _posts/2017-02-18-wordpress-elastic-beanstalk.markdown]: Dependenc**y** management ==> #dependenc**y**-management-with-composer

File: _posts/2017-02-18-wordpress-elastic-beanstalk.markdown [hunk 6]
```diff
@@ -0,0 +1,964 @@
...
+  - [Phing environment variables](#phing-environment-variables)
+  - [Configuring Wordpress](#configuring-wordpress)
+  - [Starting docker-compose](#starting-docker-compose)
+  - [Initializing the Database](#initializing-the-database)
+  - [WordPress Base Install](#wordpress-base-install)
+  - [Configuring your development environment](#configuring-your-development-environment)
+    - [Plugin Installation](#plugin-installation)
+    - [Running the behat tests](#running-behat-tests)
+    - [Themes Installation](#themes-installation)
+    - [Developing a dumy plugin](#developing-a-dumy-plugin)
```
Comment (ppaulis) [inline _posts/2017-02-18-wordpress-elastic-beanstalk.markdown]: [Developing a dummy plugin](#developing-a-dummy-plugin)

File: _posts/2017-02-18-wordpress-elastic-beanstalk.markdown [hunk 7]
```diff
@@ -0,0 +1,964 @@
...
+- [Set-up AWS Elastic BeanStalk Staging environment](#set-up-aws-elastic-beanstalk-staging-environment)
```
Comment (ppaulis) [inline _posts/2017-02-18-wordpress-elastic-beanstalk.markdown]: Bean**s**talk

File: _posts/2017-02-18-wordpress-elastic-beanstalk.markdown [hunk 8]
```diff
@@ -0,0 +1,964 @@
...
+  - [Set-up the AWS environment accounts](#set-up-the-aws-environment-accounts)
+  - [Set-up the WordPress backup S3 bucket](#set-up-the-wordpress-backup-s3-bucket)
+  - [Set-up the WordPress backup S3 bucket IAM policy](#set-up-the-wordpress-backup-s3-bucket-iam-policy)
+  - [Configuring the UpdraftPlus WordPress plugin](#configuring-the-updraftplus-wordpress-plugin)
+  - [Creating the EC2 Key Pair](#creating-the-ec2-key-pair)
+  - [Creating the MySQL DB Instance](#creating-the-mysql-db-instance)
+  - [Set-up your Elastic BeanStalk Application](#set-up-your-elastic-beanstalk-application)
```
Comment (ppaulis) [inline _posts/2017-02-18-wordpress-elastic-beanstalk.markdown]: Bean**s**talk

File: _posts/2017-02-18-wordpress-elastic-beanstalk.markdown [hunk 9]
```diff
@@ -0,0 +1,964 @@
...
+  - [Application Project Set-up in continuousphp](#application-project-set-up-in-continuousphp)
+  - [Deployment pipeline Configuration](#deployment-pipeline-configuration)
+  - [Deploying Wordpress](#deploying-wordpress)
+- [Notes](#notes)
+
+## Introduction
+
+Have you ever been scared on clicking on that WordPress or plugins update button ?
+
+Have you ever wanted to have a staging environement for the validation of your latest development by your customers and to update the production with one push upon validation ?
```
Comment (ppaulis) [inline _posts/2017-02-18-wordpress-elastic-beanstalk.markdown]: with one click

File: _posts/2017-02-18-wordpress-elastic-beanstalk.markdown [hunk 10]
```diff
@@ -0,0 +1,964 @@
...
+
+Have you ever wanted to have automated test to ensure proper site function before deployments ?
```
Comment (ppaulis) [inline _posts/2017-02-18-wordpress-elastic-beanstalk.markdown]: automated tests

File: _posts/2017-02-18-wordpress-elastic-beanstalk.markdown [hunk 11]
```diff
@@ -0,0 +1,964 @@
...
+
+The goal of this project is to have a WordPress environment that let you develop, test, package and deploy an immutable WordPress on different environments like staging and production.
```
Comment (ppaulis) [inline _posts/2017-02-18-wordpress-elastic-beanstalk.markdown]: lets

File: _posts/2017-02-18-wordpress-elastic-beanstalk.markdown [hunk 12]
```diff
@@ -0,0 +1,964 @@
...
+
+It is based on docker-compose for the local development, [continuousphp](https://continuousphp.com) for building, testing and deploying on AWS Elastic BeanStalk Infrastructure environments.
```
Comment (ppaulis) [inline _posts/2017-02-18-wordpress-elastic-beanstalk.markdown]: Bean**s**talk

File: _posts/2017-02-18-wordpress-elastic-beanstalk.markdown [hunk 13]
```diff
@@ -0,0 +1,964 @@
...
+* docker
+* docker-compose
+* composer
+* AWS account
+
+## Set-up your Development environment
+
+### Prepare your WordPress Repository
+
+Fork and clone the [WordPress Elastic BeanStalk Demo](https://github.com/continuousdemo/wordpress-eb-demo).
```
Comment (ppaulis) [inline _posts/2017-02-18-wordpress-elastic-beanstalk.markdown]: I'll willstop now highlighting the "S" in Beanstalk. Jut do a case sensitive search & replace :D

File: _posts/2017-02-18-wordpress-elastic-beanstalk.markdown [hunk 14]
```diff
@@ -0,0 +1,964 @@
...
+Because we are going to configure continuousphp in staging environment, we need to create a develop branch, for which we will create a deployment pipeline.
+
+Create the **develop** branch:
+
+```bash
+git checkout -B develop
+git push origin develop
+```
+
+This WordPress Repository include the following files:
```
Comment (ppaulis) [inline _posts/2017-02-18-wordpress-elastic-beanstalk.markdown]: includes

File: _posts/2017-02-18-wordpress-elastic-beanstalk.markdown [hunk 15]
```diff
@@ -0,0 +1,964 @@
...
+
+Install Wordpress and plugins using composer like:
+
+```
+composer.phar install
+```
+
+### Phing environment variables
+
+To set your local development environment copy the **build.local.properties.sample** to **build.local.properties**.
```
Comment (ppaulis) [inline _posts/2017-02-18-wordpress-elastic-beanstalk.markdown]: environment, copy

File: _posts/2017-02-18-wordpress-elastic-beanstalk.markdown [hunk 16]
```diff
@@ -0,0 +1,964 @@
...
+define('SECURE_AUTH_KEY',  $_SERVER["SECURE_AUTH_KEY"]);
+define('LOGGED_IN_KEY',    $_SERVER["LOGGED_IN_KEY"]);
+define('NONCE_KEY',        $_SERVER["NONCE_KEY"]);
+define('AUTH_SALT',        $_SERVER["AUTH_SALT"]);
+define('SECURE_AUTH_SALT', $_SERVER["SECURE_AUTH_SALT"]);
+define('LOGGED_IN_SALT',   $_SERVER["LOGGED_IN_SALT"]);
+define('NONCE_SALT',       $_SERVER["NONCE_SALT"]);
+```
+
+Once finished, copy the wp/wp-config.php in the wp-root/ folder, we will need it for the deployment provisioning.
```
Comment (ppaulis) [inline _posts/2017-02-18-wordpress-elastic-beanstalk.markdown]: folder. We will

File: _posts/2017-02-18-wordpress-elastic-beanstalk.markdown [hunk 17]
```diff
@@ -0,0 +1,964 @@
...
+
+Keep those keys in a safe place.
+
+### Starting docker-compose
+
+```
+docker-compose up
+```
+
+Note: if you don't daemonize the docker-compose, you will need to open a new terminal, don't forget to export again your environment variables.
```
Comment (ppaulis) [inline _posts/2017-02-18-wordpress-elastic-beanstalk.markdown]: open a new terminal. Don't forget

File: _posts/2017-02-18-wordpress-elastic-beanstalk.markdown [hunk 18]
```diff
@@ -0,0 +1,964 @@
...
+Complete the information:
+
+* Site Title: Your site title
+* Username: **admin**
+* Search Engine Visibility: Check discourage search engines from indexing this site
+* Click **Install WordPress**
+
+Your WordPress is now ready to be customized.
+
+Note: you may recieve a 500 Error because a phpmailerException, just refresh the page.
```
Comment (ppaulis) [inline _posts/2017-02-18-wordpress-elastic-beanstalk.markdown]: recieve => receive

File: _posts/2017-02-18-wordpress-elastic-beanstalk.markdown [hunk 19]
```diff
@@ -0,0 +1,964 @@
...
+"wpackagist-plugin/business-profile": "1.1.1",
+"wpackagist-plugin/updraftplus": "1.12.32",
+"wpackagist-plugin/w3-total-cache": "0.9.5.1",
+"robmorgan/phinx": "~0.6.0",
+"wp-cli/wp-cli": "^1.0",
+"phing/phing": "~2.14"
+},
+```
+
+Edit the **build.properties** and add our new plugins into the wp.plugins variables, this varibales is used by the Phing target wp-plugins-activate.
```
Comment (ppaulis) [inline _posts/2017-02-18-wordpress-elastic-beanstalk.markdown]: this varibales is used ==> one variable or multiple?

File: _posts/2017-02-18-wordpress-elastic-beanstalk.markdown (hunk 1 above)
Comment (oswaldderiemaecker) [inline _posts/2017-02-18-wordpress-elastic-beanstalk.markdown]: Removed, well seen!

**Pull request merged**